src/main/java/org/jetbrains/bsp/bazel/server/DiagnosticsDispatcher.java

```diff
@@ -0,0 +1,139 @@
+package org.jetbrains.bsp.bazel.server;
+
+import ch.epfl.scala.bsp4j.BuildClient;
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import ch.epfl.scala.bsp4j.Diagnostic;
+import ch.epfl.scala.bsp4j.DiagnosticSeverity;
+import ch.epfl.scala.bsp4j.Position;
+import ch.epfl.scala.bsp4j.PublishDiagnosticsParams;
+import ch.epfl.scala.bsp4j.Range;
+import ch.epfl.scala.bsp4j.SourceItem;
+import ch.epfl.scala.bsp4j.TextDocumentIdentifier;
+import com.google.common.collect.Lists;
+import io.bazel.rules_scala.diagnostics.Diagnostics;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.bsp.bazel.common.Uri;
+
+public class DiagnosticsDispatcher {
+
+  private static final Logger LOGGER = LogManager.getLogger(DiagnosticsDispatcher.class);
+
+  private final BazelBspServer bspServer;
+  private final BuildClient bspClient;
+
+  public DiagnosticsDispatcher(BazelBspServer bspServer, BuildClient bspClient) {
+    this.bspServer = bspServer;
+    this.bspClient = bspClient;
+  }
+
+  public void collectDiagnostics(
```

Comment:
it is connected with previous comment - this function should return this map instead of taking it as an argument